pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/analytics/StallTracker.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) SRG SSR. All rights reserved.
+ * License information is available from the LICENSE file.
+ */
+package ch.srgssr.pillarbox.player.analytics
+
+import android.util.Log
+import androidx.media3.common.MediaItem
+import androidx.media3.common.PlaybackException
+import androidx.media3.common.Player
+import androidx.media3.exoplayer.analytics.AnalyticsListener
+import androidx.media3.exoplayer.source.LoadEventInfo
+import androidx.media3.exoplayer.source.MediaLoadData
+import java.io.IOException
+import kotlin.time.Duration.Companion.milliseconds
+
+/**
+ * Stall tracker
+ * # Definition of a Stall
+ * A Stall occurs when the player is buffering during playback without user interaction.
+ */
+class StallTracker : AnalyticsListener {
+    private var stallCount = 0
+    private var lastStallTime = 0L
+    private var stallDuration = 0L
+    private var lastIsPlaying = 0L
+    private var totalPlaytimeDuration = 0L
+
+    private enum class State {
+        IDLE,
+        READY,
+        STALLED,
+        SEEKING,
+    }
+
+    private var state: State = State.IDLE
+        set(value) {
+            if (value == field) return
+            if (value == State.STALLED) {
+                lastStallTime = System.currentTimeMillis()
+                stallCount++
+            }
+            if (field == State.STALLED) {
+                stallDuration += System.currentTimeMillis() - lastStallTime
+            }
+            field = value
+        }
+
+    private fun reset() {
+        state = State.IDLE
+
+        Log.d(TAG, "Metrics: #Stalls = $stallCount duration = ${stallDuration.milliseconds} totalPlayTime = ${totalPlaytimeDuration.milliseconds}")
+        stallCount = 0
+        lastStallTime = 0L
+        stallDuration = 0
+        totalPlaytimeDuration = 0
+    }
+
+    override fun onMediaItemTransition(eventTime: AnalyticsListener.EventTime, mediaItem: MediaItem?, reason: Int) {
+        reset()
```

Comment:
Do we need to call `reset()` if the reason is `MEDIA_ITEM_TRANSITION_REASON_REPEAT`?

Reply:
Good point